### PR TITLE
chore: use infra-checkers static analysis settings and bump go to  1.18

### PR DESCRIPTION
**Pull request by sigilioso**

### CODE REVIEW

File: .github/workflows/push_pr.yml [hunk 1]
```diff
@@ -78,13 +78,18 @@ jobs:
     runs-on: ubuntu-20.04
     steps:
       - uses: actions/checkout@v3
+      - uses: actions/setup-go@v2
```
Comment (paologallinaharbur) [inline .github/workflows/push_pr.yml:81]: There is GO_VERSION env variable, can you also update it at line 60 and elsewhere?

By the way I asked in many PRs, so sorry if you read it already, why do we need to setup go now?

**Review by paologallinaharbur — APPROVED** [review]
Just a nit and small doubts regarding the way go is setup

File: .github/workflows/push_pr.yml (hunk 1 above)
Comment (sigilioso) [reply .github/workflows/push_pr.yml:81]: Sure, thanks!

Comment (sigilioso) [reply .github/workflows/push_pr.yml:81]: Including this comment to keep in handy: Needed to use golangci-lint-action v3. Review [this](https://github.com/golangci/golangci-lint-action/pull/403) for details.

**Review by paologallinaharbur — APPROVED** [review]

**Pull request merged**